state/protocol/badger/mutator_test.go

```diff
@@ -1155,3 +1201,232 @@ func TestExtendEpochTransitionWithoutCommit(t *testing.T) {
+		// bootstrap the root block
+		block1, result, seal := unittest.BootstrapFixture(participants)
+		block1.Payload.Guarantees = nil
+		block1.Header.PayloadHash = block1.Payload.Hash()
+		err := state.Mutate().Bootstrap(block1, result, seal)
+		require.NoError(t, err)
+
+		// create block2 and block3
+		block2 := unittest.BlockWithParentFixture(block1.Header)
+		block2.Payload.Guarantees = nil
+		block2.Header.PayloadHash = block2.Payload.Hash()
+		err = state.Mutate().HeaderExtend(&block2)
+		require.Nil(t, err)
+
+		block3 := unittest.BlockWithParentFixture(block2.Header)
+		block3.Payload.Guarantees = nil
+		block3.Header.PayloadHash = block3.Payload.Hash()
+		err = state.Mutate().HeaderExtend(&block3)
+		require.Nil(t, err)
+
+		// create seals for block2 and block3
+		seal2 := unittest.SealFixture(
+			unittest.SealWithBlockID(block2.ID()),
+			unittest.WithInitalState(seal.FinalState),
+		)
+		seal3 := unittest.SealFixture(
+			unittest.SealWithBlockID(block3.ID()),
+			unittest.WithInitalState(seal2.FinalState),
+		)
+
+		// include the seals in block4
+		block4 := unittest.BlockWithParentFixture(block3.Header)
+		block4.Payload.Guarantees = nil
+		block4.SetPayload(flow.Payload{
+			// placing seals in the reversed order to test
+			// Extend will pick the highest sealed block
+			Seals: []*flow.Seal{seal3, seal2},
+		})
+		block4.Header.PayloadHash = block4.Payload.Hash()
+		err = state.Mutate().HeaderExtend(&block4)
+		require.Nil(t, err)
+
+		finalCommit, err := state.AtBlockID(block4.ID()).Commit()
+		assert.NoError(t, err)
+		assert.Equal(t, seal3.FinalState, finalCommit)
+	})
+}
+
+func TestMakeValid(t *testing.T) {
+	t.Run("should trigger BlockProcessable with parent block", func(t *testing.T) {
+		consumer := &mockprotocol.Consumer{}
+
+		util.RunWithProtocolStateAndConsumer(t, consumer, func(db *badger.DB, state *protocol.State) {
+			// bootstrap the root block
+			block1, result, seal := unittest.BootstrapFixture(participants)
+			block1.Payload.Guarantees = nil
+			block1.Header.PayloadHash = block1.Payload.Hash()
+			err := state.Mutate().Bootstrap(block1, result, seal)
+			require.NoError(t, err)
+
+			// create block2 and block3
+			block2 := unittest.BlockWithParentFixture(block1.Header)
+			block2.Payload.Guarantees = nil
+			block2.Header.PayloadHash = block2.Payload.Hash()
+			err = state.Mutate().Extend(&block2)
+			require.Nil(t, err)
+
+			block3 := unittest.BlockWithParentFixture(block2.Header)
+			block3.Payload.Guarantees = nil
+			block3.Header.PayloadHash = block3.Payload.Hash()
+			err = state.Mutate().Extend(&block3)
+			require.Nil(t, err)
+
+			consumer.On("BlockProcessable", mock.Anything).Return()
+
+			// make valid on block2 and block3
+			err = state.Mutate().MarkValid(block2.ID())
+			require.NoError(t, err)
+
+			// because the parent block is the root block,
+			// BlockProcessable is not triggered on root block.
+			consumer.AssertNotCalled(t, "BlockProcessable")
```

Comment:
Added test case for verifying the event is emitted.

Reply:
Most of the PR has been reviewed already. Just need to review this test case.

@AlexHentschel @jordanschalm